.github/workflows/publish-apispec.yaml

```diff
@@ -0,0 +1,135 @@
+          echo "VERSION=${VERSION}" >> $GITHUB_ENV
+          echo "SHORT_SHA=${GITHUB_SHA::7}" >> $GITHUB_ENV
+
+      - name: Generate OpenAPI spec
+        run: |-
+          mkdir openApi-artifacts
+          dotnet test tests/Jellyfin.Server.Integration.Tests/*.csproj \
+            --filter "Jellyfin.Server.Integration.Tests.OpenApiSpecTests" \
+            --configuration Release \
+            -o openApi-artifacts \
+            --verbosity minimal \
+            --nologo
+
+      - name: Check diff to published
+        id: diff
+        run: |-
+          curl -fsSL https://api.jellyfin.org/openapi/jellyfin-openapi-${{ env.STAGE }}.json -o jellyfin-openapi-${{ env.STAGE }}.json
+          if ! diff -qiwB openApi-artifacts/openapi.json jellyfin-openapi-${{ env.STAGE }}.json &>/dev/null; then
+            echo "::set-output name=has-diff::true"
+          else
+            echo "::set-output name=has-diff::false"
+          fi
+          mv openApi-artifacts/openapi.json openApi-artifacts/jellyfin-openapi-${{ env.VERSION }}.json
+
+      - name: Publish artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: jellyfin-openapi-spec
+          path: openApi-artifacts/jellyfin-openapi-${{ env.VERSION }}.json
+
+      - name: Upload OpenAPI spec to repository server
+        uses: burnett01/rsync-deployments@4.1
+        if: ${{ needs.open-api.outputs.has-diff == 'true' }}
+        with:
+          switches: -vrptz
+          path: openApi-artifacts/jellyfin-openapi-${{ env.VERSION }}.json
+          remote_path: /srv/repository/releases/openapi/${{ env.STAGE }}
+          remote_host: ${{ secrets.DEPLOY_HOST }}
+          remote_user: ${{ secrets.DEPLOY_USER }}
+          remote_key: ${{ secrets.DEPLOY_KEY }}
+
+      - name: Create update symlink + mirrors
+        uses: appleboy/ssh-action@v0.1.4
+        if: ${{ needs.open-api.outputs.has-diff == 'true' }}
+        with:
+          host: ${{ secrets.DEPLOY_HOST }}
+          username: ${{ secrets.DEPLOY_USER }}
+          key: ${{ secrets.DEPLOY_KEY }}
+          script_stop: true
+          script: |-
+            api_root="/srv/repository/releases/openapi"
+            api_dir="${api_root}/${{ env.STAGE }}"
+            link_name="jellyfin-openapi-${{ env.STAGE }}"
+
+            if [[ -L ${api_root}/${link_name}.json ]]; then
+                rm -f ${api_root}/${link_name}_previous.json
+                mv ${api_root}/${link_name}.json ${api_root}/${link_name}_previous.json
+            fi
+            ln -s ${api_dir}/jellyfin-openapi-${{ env.VERSION }}.json ${api_root}/${link_name}.json
+
+            mirrorbits refresh
+
+  trigger-API-cis:
+    needs:
+      - open-api
+    name: Trigger API repo CI's
+    runs-on: ubuntu-latest
+    if: ${{ needs.open-api.outputs.has-diff == 'true' }}
+    continue-on-error: true
+    strategy:
+      matrix:
+        api-repo:
+          - jellyfin-client-axios
+          - jellyfin-sdk-csharp
```

Comment:
Can probably also add the other sdks?

https://github.com/jellyfin/jellyfin-sdk-swift
https://github.com/jellyfin/jellyfin-sdk-kotlin

Reply:
Yeah it would be great if we can trigger the unstable (and stable) builds instead of scheduling it. Especially the unstable builds that happen every night, even when there's no changes, would be nice to trigger from server.

Reply:
tbh these 2 where just an example and I 100% agree, however the  CI in all apiclient/sdk repos would have to support that.
Not a big deal, since I have a working prototype workflow already xD (will update and finish that when approaching the end stage of this PR)